### PR TITLE
doc(gms/scim-api): fix title and add overview

### PR DESCRIPTION
**Pull request by sid-acryl**

### CODE REVIEW

File: docs/managed-datahub/configuring-identity-provisioning-with-ms-entra.md [hunk 1]
```diff
@@ -44,12 +73,14 @@ import FeatureAvailability from '@site/src/components/FeatureAvailability';
     <img width="70%"  src="https://raw.githubusercontent.com/datahub-project/static-assets/main/imgs/scim/application-registration.png"/>
     </p>
 
-    Create three roles having `Display Name` and `Value` as mentioned below 
+    Create three roles having `Display Name` and `Value` as mentioned below
```
Comment (ethan-cartwright) [inline docs/managed-datahub/configuring-identity-provisioning-with-ms-entra.md:76]: it won't let me comment on line 70, but let's add:

d. **Create Role**: Go back to the app created in Step #1 and go to the `Provisioning` section and click on `application registration.` to create the role

Comment (ethan-cartwright) [inline docs/managed-datahub/configuring-identity-provisioning-with-ms-entra.md:76]: it won't let me inset a suggestion on line 70, but let's add:

d. **Create Role**: Go back to the app created in Step #1 and go to the `Provisioning` section and click on `application registration.` to create the role

Comment (sid-acryl) [reply docs/managed-datahub/configuring-identity-provisioning-with-ms-entra.md:76]: Done

**Review by ethan-cartwright — APPROVED** [review]
lgtm!

**Pull request merged**